custom_components/cardata/device_tracker.py

```diff
@@ -0,0 +1,129 @@
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.device_tracker import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import CardataCoordinator
+from .entity import CardataEntity
+
+PARALLEL_UPDATES = 0
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the BMW CarData tracker from config entry."""
+    domain_data = hass.data.get(DOMAIN, {})
+    runtime_data = domain_data.get(config_entry.entry_id)
+    if not runtime_data:
+        return
+    
+    coordinator = runtime_data.coordinator
+    entities: list[CardataDeviceTracker] = []
+
+    # Create device tracker for each VIN in the coordinator
+    for vin in coordinator.data.keys():
+        entities.append(CardataDeviceTracker(coordinator, vin))
+        _LOGGER.debug("Created device tracker for VIN: %s", vin)
+    
+    async_add_entities(entities)
+
+
+class CardataDeviceTracker(CardataEntity, TrackerEntity):
+    """BMW CarData device tracker."""
+
+    _attr_force_update = False
+    _attr_translation_key = "car"
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: CardataCoordinator,
+        vin: str,
+    ) -> None:
+        """Initialize the Tracker."""
+        super().__init__(coordinator, vin, "location")
+        self._attr_unique_id = f"{vin}_tracker"
+        self._unsubscribe = None
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        from homeassistant.helpers.dispatcher import async_dispatcher_connect
+        self._unsubscribe = async_dispatcher_connect(
+            self.hass,
+            self._coordinator.signal_update,
+            self._handle_update,
+        )
+        self._handle_update(self.vin, "location")
+
+    async def async_will_remove_from_hass(self) -> None:
+        """When entity is removed from hass."""
+        await super().async_will_remove_from_hass()
+        if self._unsubscribe:
+            self._unsubscribe()
+            self._unsubscribe = None
+
+    def _handle_update(self, vin: str, descriptor: str) -> None:
+        """Handle updates from coordinator."""
+        if vin != self.vin:
+            return
+        # Update location data when any location-related descriptor changes
+        if any(loc_desc in descriptor for loc_desc in ["navigation.currentLocation", "location", "position", "gps"]):
+            self.schedule_update_ha_state()
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return entity specific state attributes."""
+        attrs = {}
```

Comment:
I would add the `vehicle_basic_data` (and `_raw`) in this extra attributes entity and remove them [to all others entities](https://github.com/JjyKsi/bmw-cardata-ha/blob/208626bf22fc103592a46a6a265b2607f61fa21e/custom_components/cardata/entity.py#L63), what do you think ?